src/redisgears_python.c

```diff
@@ -3034,6 +3034,76 @@ static PyObject* modelRunnerRun(PyObject *cls, PyObject *args){
     return tensorList;
 }
 
+static void FinishAsyncModelRun(RAI_OnFinishCtx *onFinishCtx, void *private_data) {
+
+	RAI_Error *error;
+	RedisAI_InitError(&error);
+	RAI_ModelRunCtx* mctx = RedisAI_GetAsModelRunCtx(onFinishCtx, error);
+
+	RedisGearsPy_LOCK
+	PyObject* future = private_data;
+	PyObject* l = PyList_New(0);
+	PyObject* tensorList = PyList_New(0);
+	PyObject* errs = PyList_New(0);
+
+	PyList_Append(l, tensorList);
+	PyList_Append(l, errs);
+
+	if (RedisAI_GetErrorCode(error) != RedisAI_ErrorCode_OK) {
+		const char *errStr = RedisAI_GetError(error);
+		PyObject* pyErr = PyUnicode_FromStringAndSize(errStr, strlen(errStr));
+		PyList_Append(errs, pyErr);
+		Py_DECREF(pyErr);
+		goto finish;
```

Comment:
So if there are errors there is no way there will also be results? If so maybe we should expose `set_exception` (https://docs.python.org/3/library/asyncio-future.html) and expose it to C just like we expose `setFutureResultsFunction`

If there is a possibility for results and errors then this code ignores results when there is an error.